graylog2-web-interface/src/components/event-definitions/event-definition-form/EventConditionForm.tsx

```diff
@@ -0,0 +1,154 @@
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React from 'react';
+import { PluginStore } from 'graylog-web-plugin/plugin';
+import lodash from 'lodash';
+
+import { defaultCompare as naturalSort } from 'logic/DefaultCompare';
+import { Select } from 'components/common';
+import { Clearfix, Col, ControlLabel, FormGroup, HelpBlock, Row } from 'components/bootstrap';
+import { HelpPanel } from 'components/event-definitions/common/HelpPanel';
+
+import styles from './EventConditionForm.css';
+
+import commonStyles from '../common/commonStyles.css';
+
+type Props = {
+  eventDefinition: any,
+  currentUser: any,
+  validation: any,
+  onChange: (name: string, obj: any) => void,
+  action: string,
+};
+
+const EventConditionForm = ({ eventDefinition, currentUser, validation, onChange, action }: Props) => {
+  const getConditionPlugin = (type: string | undefined) => {
+    if (type === undefined) {
+      return undefined;
+    }
+
+    return PluginStore.exports('eventDefinitionTypes').find((edt) => edt.type === type) || undefined;
+  };
+
+  const sortedEventDefinitionTypes = () => {
+    return PluginStore.exports('eventDefinitionTypes').sort((edt1, edt2) => {
+      // Try to sort by given sort order and displayName if possible, otherwise do it by displayName
+      const edt1Order = edt1.sortOrder;
+      const edt2Order = edt2.sortOrder;
+
+      if (edt1Order !== undefined || edt2Order !== undefined) {
+        const sort = lodash.defaultTo(edt1Order, Number.MAX_SAFE_INTEGER) - lodash.defaultTo(edt2Order, Number.MAX_SAFE_INTEGER);
+
+        if (sort !== 0) {
+          return sort;
+        }
+      }
+
+      return naturalSort(edt1.displayName, edt2.displayName);
+    });
+  };
+
+  const formattedEventDefinitionTypes = () => {
+    return sortedEventDefinitionTypes()
+      .map((type) => ({ label: type.displayName, value: type.type }));
+  };
+
+  const handleEventDefinitionTypeChange = (nextType) => {
+    const conditionPlugin = getConditionPlugin(nextType);
+    const defaultConfig = conditionPlugin?.defaultConfig || {};
+
+    onChange('config', { ...defaultConfig, type: nextType });
+  };
+
+  const renderConditionTypeDescriptions = () => {
+    const typeDescriptions = sortedEventDefinitionTypes()
+      .map((type) => {
+        return (
+          <React.Fragment key={type.type}>
+            <dt>{type.displayName}</dt>
+            <dd>{type.description || 'No description available.'}</dd>
+          </React.Fragment>
+        );
+      });
+
+    return <dl>{typeDescriptions}</dl>;
+  };
+
+  const disabledSelect = () => {
+    return !formattedEventDefinitionTypes().some((edt) => eventDefinition.config.type === edt.value) && action === 'edit';
+  };
+
+  const eventDefinitionType = getConditionPlugin(eventDefinition.config.type);
+
+  const eventDefinitionTypeComponent = eventDefinitionType?.formComponent
+    ? React.createElement(eventDefinitionType.formComponent, {
```

Comment:
For some reason I had to type `createElement` and ended up typing it like this, because I could not figure out a better solution:
`React.createElement<React.ComponentProps<any>>`